### PR TITLE
Rework Placements Support Org Controller to work with form objects

### PR DESCRIPTION
**Pull request by JamieCleare2525**
## Context

To continue with our mindset of using form objects when not working with actual models/records, refactor the `Placements::Support::OrganisationsController` to work an `OrganisationOnboardingForm`

## Changes proposed in this pull request

Introduce an `OrganisationOnboardingForm` to work when selecting the type on organisation which is being on boarded.

## Guidance to review

- [ ] Selecting "ITT Provider" should redirect the user to the Provider onboarding page.
- [ ] Selecting "School" should redirect the user to the School onboarding page.
- [ ] Not selection an Organisation Type should show the error "Select an organisation type"

Screenshot 2024-01-11 at 13 07 22

## Link to Trello card

https://trello.com/c/Z0wd50sA/113-refactor-support-organisation-controller-to-use-organisationonboardingform-object

### CODE REVIEW

File: app/controllers/placements/support/organisations_controller.rb [hunk 1]
```diff
@@ -5,39 +5,27 @@ def index
...
-  def organisation_types
-    @organisation_types ||= [
-      OpenStruct.new(id: "itt_provider", name: t("itt_provider")),
-      OpenStruct.new(id: "school", name: t("school")),
-    ]
-  end
-
   def organisation
     @organisation ||= Placements::School.new
   end
```
Comment (ollietreend) [inline app/controllers/placements/support/organisations_controller.rb:26]: Is this safe to remove now, too?

Comment (JamieCleare2525) [reply app/controllers/placements/support/organisations_controller.rb:26]: I guess it is :D

Comment (JamieCleare2525) [reply app/controllers/placements/support/organisations_controller.rb]: resolved in [75a5b28](https://github.com/DFE-Digital/itt-mentor-services/pull/107/commits/75a5b28cb435fe5668e151a4db38c61e2cdc4ef5)

**Review by ollietreend — APPROVED** [review]
Chef kiss emoji

**Review by elceebee — APPROVED** [review]
Beautiful

**Pull request merged**